editor/src/clj/editor/gui.clj

```diff
@@ -1389,21 +1389,30 @@
                          project (project/get-project basis self)
                          current-scene (g/node-feeding-into basis self :template-resource)]
                      (concat
-                       (if current-scene
-                         (g/delete-node current-scene)
-                         [])
-                       (if (and new-value (:resource new-value))
+                       (when current-scene
+                         (g/delete-node current-scene))
+                       (when (and new-value (:resource new-value))
                          (when-some [{connect-tx-data :tx-data
                                       scene-node :node-id
                                       created-in-tx :created-in-tx} (project/connect-resource-node evaluation-context project (:resource new-value) self [])]
+                           ;; Note: If the node was created in this transaction,
+                           ;; it means it was a missing resource node. Such a
+                           ;; node is marked defective, and we will not try to
+                           ;; override its properties.
                            (concat
                              connect-tx-data
-                             (let [properties-by-node-id (comp (or (:overrides new-value) {})
-                                                               (into {}
-                                                                     (map (fn [[k v]] [v k]))
-                                                                     (if created-in-tx
-                                                                       {}
-                                                                       (g/node-value scene-node :node-ids evaluation-context))))]
+                             (let [properties-by-node-id
+                                   (or (when-not created-in-tx
+                                         (when-let [properties-by-id (coll/not-empty (:overrides new-value))]
+                                           (let [node-ids-by-id (g/node-value scene-node :node-ids evaluation-context)]
+                                             (when (and (not (g/error-value? node-ids-by-id))
+                                                        (coll/not-empty node-ids-by-id))
+                                               (comp properties-by-id
+                                                     (into {}
+                                                           (map (fn [[id node-id]]
+                                                                  (pair node-id id)))
```

Comment:
I was about to suggest using `flipped-pair`, but then I realized it's 2-arity, while this function is 1-arity. Maybe we could add 1-arity variant of `flipped-pair`?